### PR TITLE
John's changes sans the SR conflicts

### PR DESCRIPTION
**Pull request by Contentmaudlin**

### CODE REVIEW

File: src/frontend/parser.mly [hunk 1]
```diff
@@ -253,13 +253,13 @@ array_lit:
...
-| LDBRACK list_lit_elements RDBRACK     { } 
-| LPAREN RPAREN                         { LitVector([]) }
+| LT list_lit_elements GT        { LitVector($2) }
+/*| LDBRACK list_lit_elements RDBRACK     { } */  
+| LT GT                         { LitVector([]) }
 
 list_lit_elements:
-| list_lit_elements COMMA expr          { $3::$1 }
-| expr                                  { [$1] }
+| list_lit_elements COMMA shift_expr          { $3::$1 }
```
Comment (j-hui) [inline src/frontend/parser.mly]: but what if we wanted to something like
```
[true, false, 3<4]
```
an array literal of booleans

Comment (Contentmaudlin) [reply src/frontend/parser.mly]: "what if I wanted to be a bad programmer"

Comment (Contentmaudlin) [reply src/frontend/parser.mly]: jk

Comment (Contentmaudlin) [reply src/frontend/parser.mly]: The problem with that is vector literals. What if you wanted to have a vector of booleans?

`<true, false, 3>4>`

Good luck parsing that. Obviously we designed our vectors to only allow scalars but the grammar doesn't reflect that.

File: src/frontend/parser.mly [hunk 2]
```diff
@@ -252,13 +252,16 @@ array_lit:
 | LBRACK RBRACK                         { LitArray([]) }
 
 vector_lit:
-| LT list_lit_elements GT        { LitVector($2) }
-/*| LDBRACK list_lit_elements RDBRACK     { } */  
+| LT vlist_lit_elements GT        { LitVector($2) }
```
Comment (j-hui) [inline src/frontend/parser.mly]: try changing this to `LPAREN vlist_lit_elements RPAREN`? Does that help?

**Pull request closed without merging**